dll/win32/browseui/shellbrowser.cpp

```diff
@@ -806,14 +806,28 @@ HRESULT CShellBrowser::BrowseToPIDL(LPCITEMIDLIST pidl, long flags)
     CComPtr<IShellFolder>                   newFolder;
     FOLDERSETTINGS                          newFolderSettings;
     HRESULT                                 hResult;
+    IPersist                                *pp;
+    BOOL                                    HasIconViewType;
 
     // called by shell view to browse to new folder
     // also called by explorer band to navigate to new folder
     hResult = SHBindToFolder(pidl, &newFolder);
     if (FAILED_UNEXPECTEDLY(hResult))
         return hResult;
+    hResult = newFolder->QueryInterface(IID_PPV_ARG(IPersist, &pp));
+    if (SUCCEEDED(hResult))
+    {
+        CLSID pclsid;
+        hResult = pp->GetClassID(&pclsid);
+        pp->Release();
```

Comment:
There is a helper function for this: `IUnknown_GetClassID`, that also handles another case where `IPersistFolder` is exported, but `IPersist` is not.

Reply:
Updated to use ```IUnknown_GetClassID```